llarp/dns/server.cpp

```diff
@@ -96,20 +93,13 @@ namespace llarp
     bool
     Proxy::SetupUnboundResolver(const std::vector<IpAddress>& resolvers)
     {
-      auto failFunc = [self = weak_from_this()](SockAddr to, Message msg) {
-        auto this_ptr = self.lock();
-        if (this_ptr)
-        {
-          this_ptr->SendServerMessageTo(to, std::move(msg));
-        }
+      auto failFunc = [self = shared_from_this()](SockAddr to, Message msg) {
```

Comment:
Any particular reason for this change?

Reply:
Ah, I see the commit message for this now.  I initially used weak_ptr to avoid complexity/leaks when re-initializing upstream resolvers, but if this works it's fine by me.